hypothesis-python/src/hypothesis/internal/conjecture/provider_conformance.py

```diff
@@ -0,0 +1,492 @@
+
+            try:
+                choice = draw_func(**constraints)
+                note(f"drew {choice_type} {choice}")
+                expected_type = {
+                    "integer": int,
+                    "float": float,
+                    "bytes": bytes,
+                    "string": str,
+                    "boolean": bool,
+                }[choice_type]
+                assert isinstance(choice, expected_type)
+                assert choice_permitted(choice, constraints)
+            except context_manager_exceptions as e:
+                note(f"caught exception {type(e)} in context_manager_exceptions: {e}")
+                try:
+                    self.context_manager.__exit__(type(e), e, None)
+                except BackendCannotProceed:
+                    self.frozen = True
+                    return None
+
+            return choice
+
+        @precondition(lambda self: not self.frozen)
+        @rule(constraints=integer_constraints())
+        def draw_integer(self, constraints):
+            self._draw("integer", constraints)
+
+        @precondition(lambda self: not self.frozen)
+        @rule(constraints=float_constraints())
+        def draw_float(self, constraints):
+            self._draw("float", constraints)
+
+        @precondition(lambda self: not self.frozen)
+        @rule(constraints=bytes_constraints())
+        def draw_bytes(self, constraints):
+            self._draw("bytes", constraints)
+
+        @precondition(lambda self: not self.frozen)
+        @rule(constraints=string_constraints())
+        def draw_string(self, constraints):
+            self._draw("string", constraints)
+
+        @precondition(lambda self: not self.frozen)
+        @rule(constraints=boolean_constraints())
+        def draw_boolean(self, constraints):
+            self._draw("boolean", constraints)
+
+        @precondition(lambda self: not self.frozen)
+        @rule(label=st.integers())
+        def span_start(self, label):
+            self.provider.span_start(label)
+
+        @precondition(lambda self: not self.frozen)
+        @rule(discard=st.booleans())
+        def span_end(self, discard):
+            self.provider.span_end(discard)
+
+        @precondition(lambda self: not self.frozen)
+        @rule()
+        def freeze(self):
+            # phase-transition, mimicking data.freeze() at the end of a test case.
+            self.frozen = True
+            self.context_manager.__exit__(None, None, None)
+
+        @precondition(lambda self: self.frozen)
+        @rule(
+            value=st.integers()
+            | st.floats(allow_nan=False)
+            | st.booleans()
+            | st.binary()
+            | st.text()
+        )
+        def realize(self, value):
+            # Hypothesis can in theory pass values of any type to `realize`,
+            # but st.from_type(object) acts too much like a fuzzer for crosshair
+            # internals here and finds very strange errors.
```

Comment:
I get why this is tempting, but also... isn't surfacing issues in the providers the whole point of this function?

As a pragmatic compromise, I suggest adding a `_realizable_objects=` strategy to the params (note leading underscore, to indicate private/unstable/disrecommended nature), which we simply pass here.   We can default that to `from_type(object) | from_type(type).flatmap(from_type)` for maximum diversity; and Crosshair can dial that down - or upweight the symbolic inputs, or just opt in and get fuzzed.